dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java

```diff
@@ -312,62 +440,150 @@ void deleteSecret(final DeleteSecretForm form, final User user)
+                final boolean bool = (asString.equalsIgnoreCase(Boolean.TRUE.toString())
+                        || asString.equalsIgnoreCase(Boolean.FALSE.toString()));
+                if (!bool) {
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Can not convert value `%s` to type BOOL for param `%s`.",
+                                    asString, describedParamName
+                            )
+                    );
+                }
+            }
+        }
+
+        if (!appDescriptor.isAllowExtraParameters()) {
+            final SetView<String> extraParamsFound = Sets
+                    .difference(params.keySet(), appDescriptorParams.keySet());
+
+            if (!extraParamsFound.isEmpty()) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Unknown additional params `%s` not allowed by the app descriptor.",
+                                String.join(", ", extraParamsFound)
+                        )
+                );
             }
+        }
+        return params;
+    }
+
+    /**
+     * This method is meant to validate inputs for an update that can be performed on individual
+     * properties. It assumes there's an instance already saved and only performs validations on the
+     * new incoming params. This gives the flexibility to modify the value on individual properties
+     * that are already saved. If the app We're not expecting
+     * @throws IllegalArgumentException This will give back an exception if you send an invalid param.
+     */
+    private Map<String, Input> validateFormForUpdate(final SecretForm form,
+            final AppDescriptor appDescriptor)
+            throws IllegalArgumentException {
+
+        final Map<String, Input> params = form.getInputParams();
+        if (!UtilMethods.isSet(params)) {
+            throw new IllegalArgumentException("Required Params aren't set.");
+        }
 
-            final Param incomingParam = incomingParamEntry.getValue();
-            //We revise the incoming param against the definition loaded from the yml.
-            if(describedParam.isRequired() && UtilMethods.isNotSet(incomingParam.getValue())){
-               throw new IllegalArgumentException(
-               String.format("Params named `%s` is marked as required in the descriptor but does not have any value.", incomingParamName));
+        //Param/Property names are case sensitive.
+        final Map<String, ParamDescriptor> appDescriptorParams = appDescriptor.getParams();
+        for (final Entry<String, Input> entry : params.entrySet()) {
+            final String paramName = entry.getKey();
+            final ParamDescriptor paramDescriptor = appDescriptorParams.get(paramName);
+            if (null == paramDescriptor && !appDescriptor.isAllowExtraParameters()) {
```

Comment:
this one shouldn't be an OR?

Reply:
I'm testing that the paramName is something that isn't defined on the yml and that at the same time the yml does not allow additional params. If so then we throw a validation exception.

And OR would say that the param does not exist or the yml does not allow extra params. 
iI think it is ok like that.